docs/howtos/numpy_eigensolver.rst

```diff
@@ -0,0 +1,51 @@
+.. _how-to-numpy:
+
+How to use the ``NumPyEigensolver``
+===================================
+
+In order to ensure a physically meaningful excited states of a hamiltonian are found when using the
+:class:`~qiskit.algorithms.minimum_eigensolvers.NumPyEigensolver` one needs to set the
+:attr:`~qiskit.algorithms.minimum_eigensolvers.NumPyEigensolver.filter_criterion` attribute
+of the solver.
+
+Subclasses of :class:`~qiskit_nature.second_q.problems.BaseProblem` in Qiskit Nature provide the
+:meth:`~qiskit_nature.second_q.problems.BaseProblem.get_default_filter_criterion` method which
+provides a default implementation of such a filter criterion for commonly encountered cases.
+
+Below we show how you can use this setting.
+
+1. We obtain an ``ElectronicStructureProblem`` which we want to solve:
+
+.. testcode::
+
+   from qiskit_nature.second_q.drivers import PySCFDriver
+   driver = PySCFDriver(atom="H 0 0 0; H 0 0 0.735", basis="sto-3g")
+   problem = driver.run()
+
+2. We setup our ``QubitMapper``:
+
+.. testcode::
+
+    from qiskit_nature.second_q.mappers import JordanWignerMapper
+    mapper = JordanWignerMapper()
+
+3. We setup our ``NumPyEigensolver``:
+
+.. testcode::
+
+    from qiskit.algorithms.eigensolvers import NumPyEigensolver
+    algo = NumPyEigensolver(k=100)
+    algo.filter_criterion = problem.get_default_filter_criterion()
+
+4. We wrap everything in a ``ExcitedStatesEigensolver``:
+
+.. testcode::
+
+    from qiskit_nature.second_q.algorithms import ExcitedStatesEigensolver
+    solver = ExcitedStatesEigensolver(mapper, algo)
+
+5. We solve the problem:
+
+.. testcode::
+
+    result = solver.solve(problem)
```

Comment:
I am thinking for code like this, which should run fast, can we not do testoutput here so we can keep an eye on the howto that it continues to work as expected. I thought about commenting on the AdaptVQE one but am not sure how long that takes to run. If its relatively quick though...

Reply:
Alright I added assertions against the total energies up to 4 decimal places. Two comments:
1. I cannot test the UVCC case because I do not actually have a valid problem loaded to test there
2. I chose to actually include the output in the rendering because otherwise I think it can look confusing with the `print` statement included but no output actually shown